app/lib/search/backend.dart

```diff
@@ -427,17 +428,40 @@ class SearchBackend {
     }
   }
 
-  /// Downloads the remote SDK content relative to the base URI.
-  Future<String> fetchSdkIndexContentAsString({
-    required Uri baseUri,
-    required String relativePath,
+  /// Downloads the remote SDK content from [uri] and creates a cached file in the
+  /// `.dart_tool/pub-search-data/` directory.
+  ///
+  /// When a local file in `app/.dart_tool/pub-search-data/<reduced-uri>` exists,
+  /// its content will be loaded instead. URL reduction replaces slashes and other
+  /// non-characters with a single dash `-`, like:
+  /// - `https-api.dart.dev-stable-latest-index.json`
+  Future<String> loadOrFetchSdkIndexJsonAsString(
+    Uri uri, {
+    @visibleForTesting Duration? ttl,
   }) async {
-    final uri = baseUri.resolve(relativePath);
+    final fileName = uri.toString().replaceAll(RegExp(r'[^a-z0-9\.]+'), '-');
+    final file = File(p.join('.dart_tool', 'pub-search-data', fileName));
+    if (await file.exists()) {
```

Comment:
It's always safer to read and then handle the `IOException` if the file doesn't exist.

Reply:
Right. As the same thing can be said of the `_http.get` that this method is also using, I'll address that try-catch in one level up, in a follow-up PR.

Reply:
Note: the SDK index builder methods (both for Dart and Flutter) already have a try-catch block, handling exceptions.